src/YMenuBar.h

```diff
@@ -0,0 +1,230 @@
+  is distributed in the hope that it will be useful, but WITHOUT ANY
+  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+  FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+  License for more details. You should have received a copy of the GNU
+  Lesser General Public License along with this library; if not, write
+  to the Free Software Foundation, Inc., 51 Franklin Street, Fifth
+  Floor, Boston, MA 02110-1301 USA
+*/
+
+
+/*-/
+
+  File:		YMenuBar.h
+
+  Author:	Stefan Hundhammer <shundhammer@suse.de>
+
+/-*/
+
+#ifndef YMenuBar_h
+#define YMenuBar_h
+
+#include "YSelectionWidget.h"
+#include "YMenuItem.h"
+
+class YMenuBarPrivate;
+
+
+/**
+ * A classical menu bar for pulldown menus.
+ *
+ * Use this only when appropriate: In most places, YaST follows a wizard-driven
+ * UI strategy, asking the user (ideally) one question at a time, with a [Next]
+ * and a [Back] button to move between wizard steps. One of the last steps is
+ * usually presenting the collected information to the user until everything is
+ * applied.
+ *
+ * A menu bar OTOH is meant for the opposite UI strategy: Presenting some kind
+ * of document (which may also be a number of input fields) to the user as the
+ * central point and providing lots of different operations on that document.
+ * The two concepts don't mix very well, so use this widget with caution.
+ *
+ * A menu bar should only contain menus, no direct actions.
+ **/
+class YMenuBar: public YSelectionWidget
+{
+protected:
+    /**
+     * Constructor.
+     **/
+    YMenuBar( YWidget * parent );
+
+public:
+    /**
+     * Destructor.
+     **/
+    virtual ~YMenuBar();
+
+    /**
+     * Returns a descriptive name of this widget class for logging,
+     * debugging etc.
+     **/
+    virtual const char * widgetClass() const { return "YMenuBar"; }
+
+    /**
+     * Rebuild the displayed menu tree from the internally stored YMenuItems.
+     *
+     * The application should call this (once) after all items have been added
+     * with addItem(). YMenuButton::addItems() calls this automatically.
+     *
+     * Derived classes are required to implement this.
+     **/
+    virtual void rebuildMenuTree() = 0;
+
+    /**
+     * Add multiple items. For some UIs, this can be more efficient than
+     * calling addItem() multiple times. This function also automatically calls
+     * resolveShortcutConflicts() and rebuildMenuTree() at the end.
+     *
+     * Derived classes can overwrite this function, but they should call this
+     * base class function at the end of the new implementation.
+     *
+     * Reimplemented from YSelectionWidget.
+     **/
+    virtual void addItems( const YItemCollection & itemCollection );
+
+    /**
+     * Add one item. This widget assumes ownership of the item object and will
+     * delete it in its destructor.
+     *
+     * This reimplementation will an index to the item that is unique for all
```

Comment:
Will what?

Reply:
This was an intermediate state of that file that is now outdated. Please check the latest version.